packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -191,18 +197,22 @@ export function getLargestContentfulPaint(metric: Metric): ReplayPerformanceEntr
  * Add a CLS event to the replay based on a CLS metric.
  */
 export function getCumulativeLayoutShift(metric: Metric): ReplayPerformanceEntry<WebVitalData> {
-  const lastEntry = metric.entries[metric.entries.length - 1] as
-    | (PerformanceEntry & { sources?: LayoutShiftAttribution[] })
-    | undefined;
-  const nodes: Node[] = [];
-  if (lastEntry && lastEntry.sources) {
-    for (const source of lastEntry.sources) {
-      if (source.node) {
-        nodes.push(source.node);
+  const layoutShifts = [];
+  for (const entry of metric.entries) {
+    const layoutShift = entry as LayoutShift | undefined;
+    if (layoutShift) {
+      const sources = [];
+      for (const source of layoutShift.sources) {
+        const nodeId = record.mirror.getId(source.node);
+        if (nodeId) {
+          sources.push(nodeId);
+        }
       }
+      layoutShifts.push({ value: layoutShift.value, sources: sources.length ? sources : undefined });
```

Comment:
Is it ok if sources is just an empty array?

```suggestion
      layoutShifts.push({ value: layoutShift.value, sources });
```